### PR TITLE
Prepare for Post Editor metadata Reduxification

### PR DESCRIPTION
**Pull request by jsnajdr**
This PR is a series of three refactorings that don't change any behavior, but will make future PRs from the "Post Editor Metadata" series smaller and easier to review.

### Post Editor Reducer: refactor receiving post updates

Extract the code that computes unapplied edits into a separate variable assignment. It will be useful when making future changes related to metadata editing.

### Post Editor Reducer: use `mergeIgnoringArrays` only to merge actual post edits

Don't use `mergeIgnoringArrays` to do the `keyedReducer`-like job of updating the `siteId` and `postId` nested state. The function is soon going to be incompatible with that and capable only of correctly merging actual post edits.

### Post Editor: rename and split `mergeIgnoringArrays` into two functions

Rename the `mergeIgnoringArrays` to a more appropriate name: `mergePostEdits`. It's a function that merges two post edits into one. It's used by the `edits` reducer to merge existing and incoming post edits.

Also create a function called `applyPostEdits` that applies a post edits object to a post object and returns a modified post. It's used by the `getEditedPost` selector that takes an original post, a post edits object and returns a modified post.

At this moment, both implementations are identical, but that will change soon when editing post metadata is intruduced into the Post Editor Redux store.

### How to test

Covered by unit tests. In Calypso UI, test that editing posts is not broken.

Part of #24281

### CODE REVIEW

**Comment by jsnajdr** [comment]
> What do you think if we removed applyPostEdits and used mergePostEdits in all cases for now, and added back the updated version of applyPostEdits when it becomes needed?

@copons Good idea -- I pushed a commit where `applyPostEdits` is removed for now. When working on #24291 earlier today, I realized that it might not be needed after all and that one `mergePostEdits` could be sufficient. I'll verify that later.

I also added some comments to the `posts.edits` reducer. Let me know if that helps or if there's some other code that needs explaining. The Post Editor is very complex and not easy to understand.

**Comment by jsnajdr** [comment]
> I think it's clear enough right now (for the Post Editor standards, at least 😄)

The good news is that most of the complexity seems to be coming from the Flux/Redux interactions and syncing stuff between the two stores. Once everything is in Redux, the editor should get much simpler.

@Copons Are there any other concerns to look at before approving and merging this PR?

**Comment by Copons** [comment]
> Are there any other concerns to look at before approving and merging this PR?

@jsnajdr Not at all, but I assumed you were waiting for a @Automattic/team-calypso green light. 🙂

**Review by Copons — APPROVED** [review]

**Pull request merged**
